app/jobs/attach_files_to_work_job.rb

```diff
@@ -14,6 +14,7 @@ def perform(work, uploaded_files)
       attach_content(actor, uploaded_file.file)
       uploaded_file.update(file_set_uri: file_set.uri)
     end
+    InheritPermissionsJob.perform_later(work)
```

Comment:
you could set the permissions on the new file set when it's created. This is how it's being done with visibility a few lines above.